### PR TITLE
Remove duplicated tests in Keras mixins

### PR DESCRIPTION
**Pull request by Wauplin**
In the keras tests, `HubKerasFunctionalTest` was inheriting from `HubKerasSequentialTest` itself inheriting from `HubMixingTestKeras`. Inheritance was useful to get `setUpClass` and `tearDown` initializers in all 3 classes. However, test cases were also all inherited which duplicated them at runtime (some of them being ran 3 times exactly with the same parameters).

This PR makes the inheritance from a common parent class. Number of test cases went from 44 to 22, reducing redundancy.

**EDIT:** some others tests seem to be duplicated as well (source code itself is copy-pasted) but addressing this is not the purpose of this PR.

### CODE REVIEW

**Comment by osanseviero** [comment]
LGTM but I would love to wait for @merveenoyan to review it as well :)

**Comment by Wauplin** [comment]
Yes sure, it's not urgent anyway.

**Review by LysandreJik — APPROVED** [review]
Would also wait for Merve's approval before merging, but LGTM on my side :+1:.

Thank you for taking care of it!

File: tests/test_keras_integration.py
Comment (merveenoyan) [inline tests/test_keras_integration.py:538]: Thanks for this catch! Looks neat

File: tests/test_keras_integration.py
Comment (merveenoyan) [inline tests/test_keras_integration.py:546]: What do you say if we take this outside given it just doesn't matter according to how model is built? (instead of making this a class specific method)

**Review by merveenoyan — APPROVED** [review]
I left a minor comment, feel free to disregard if you think your way is better 🙂 (it's a minor nit to avoid repetition)

File: tests/test_keras_integration.py
Comment (Wauplin) [reply tests/test_keras_integration.py:546]: I thought about that but it's maybe preferable to keep it close to each `model_init` so that they can evolve together. Here the fact that `model_fit` is the same is just "by chance" that both `model_init` takes the same input, right ?

Comment (Wauplin) [reply tests/test_keras_integration.py:546]: I'm totally fine with putting it outside though. Your call !

Comment (merveenoyan) [reply tests/test_keras_integration.py:546]: Sure, let's keep it your way.

**Pull request merged**
